### DISCUSSION TITLE
created whl data cannot open by zipfile.py

### DISCUSSION THREAD

**Discussion by grekichi** [pypa/setuptools#4408] [Q&A]
Hi there !
I need someone's help about wheel data created by setuptools.
I try to create whl data. (ctc-segmentation produced by [lumaku](https://github.com/lumaku/ctc-segmentation))

Build succeeded but that file cannot open by zipfile.py.
Build command is `python -m build --wheel `

Error message is below,
```
File "/Users/bjc-o/miniforge3/envs/app_ml/lib/python3.10/zipfile.py", line 1334, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file 
```

----------------------------
Development environment
---------------------------
OS: M1 macOS Monterey 12.7.5
Python ver: 3.10.9
Setuptools==70.0.0 (also tried 69.4.0)
wheel==0.43.0
cython==0.29.36
numpy==1.26.4
----------------------------

Ceated data for build whl is here.
https://github.com/grekichi/p4a_wheel/blob/main/ctc_segmentation-1.7.4-cp310-cp310-macosx_11_0_arm64.whl

This data can open 'The Unarchiver' app.

Prepared each another data by build as follows.

----------
### setup.py
----------------------------------------------------------------------------------------------
```
from setuptools import setup, find_packages, Extension
from setuptools.command.build_ext import build_ext
import numpy


try:
    from Cython.Build import cythonize
    USE_CYTHON = True
except ImportError:
    USE_CYTHON = False

'# https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html'
ext = '.pyx' if USE_CYTHON else '.c'
extensions = [
    Extension(
        name="ctc_segmentation.ctc_segmentation_dyn",
        sources=["ctc_segmentation/ctc_segmentation_dyn"+ext],
        include_dirs=[numpy.get_include()],
    )
]
if USE_CYTHON:
    from Cython.Build import cythonize
    extensions = cythonize(extensions)

package_information = """
'# CTC segmentation'

CTC segmentation is used to align utterances within audio files.
It can be combined with CTC-based ASR models.
This package includes the core functions.

https://github.com/lumaku/ctc-segmentation
"""

setup(
    name="ctc_segmentation",
    version="1.7.4",
    python_requires='>=3.6',
    packages=find_packages(),
    setup_requires=["numpy"],
    install_requires=["setuptools", "numpy", "Cython"],
    zip_safe=False,
    ext_modules=extensions,
    cmdclass={'build_ext': build_ext},

    author="Ludwig Kuerzinger <ludwig.kuerzinger@tum.de>, "
           "Dominik Winkelbauer <dominik.winkelbauer@tum.de>",
    description="CTC segmentation to align utterances within "
                "large audio files.",

    long_description_content_type="text/markdown",
    long_description=package_information,
)
```
-----------------------------------------------------------------------------------------------

------------------
### project.toml
--------------------------------------------------------
```
[build-system]
requires = ["wheel", "setuptools", "numpy", "Cython"]
build-backend = "setuptools.build_meta"
```
---------------------------------------------------------

-------------------
### MANIFEST.in
------------------------------------------------------------
```
include MANIFEST.in
include LICENSE.txt
Include pyproject.toml
include setup.py
include ctc_segmentation/ctc_segmentation.py
include ctc_segmentation/ctc_segmentation_dyn.pyx
include ctc_segmentation/__init__.py
include ctc_segmentation/partitioning.py
```
-------------------------------------------------------------

I cannot discover the error cause.
If you have any advice, please tell me.

Thanks in advance.

**Comment by abravalheri** [comment]
Hi @grekichi, this might be related to the specific compiler in macOS, or a particular binary having an anomalous invalid bit sequence somewhere...
Unfortunately I don't have access to a macOS system to try reproducing the error and investigating deeper.

I suggest you trying to build it again without the `--wheel` flag (adding `--wheel` is a "shortcut"... the most consistent results are the ones obtained when you build the sdist first and then the wheel from the sdist -- that is the detault behaviour of `build` when you don't pass other flags).

**Reply by grekichi** [reply]
Hi Anderson, thanks for your advice.

I tried your recommend command `python -m build` , but result is same. return error: "File is not a zip file."
Also I tried `python setup.py sdist` , `python setup.py sdist bdist_wheel` and `python setup.py bdist_wheel`.
All failed 😭 

In case of windows or linux, will it be not occur this kinds of trouble ?

**Reply by abravalheri** [reply]
> In case of windows or linux, will it be not occur this kinds of trouble ?

This might be something worth for you to check. Let me know if you find a problem there too.

Unfortunately I will not be able to help on macOS.

**Reply by grekichi** [reply]
I discovered the main cause is wrong url.
reference site:
https://github.com/samjabrahams/tensorflow-on-raspberry-pi/issues/24

**cannot zip open:**
`https://github.com/grekichi/p4a_wheel/blob/main/.github/ctc_segmentation-1.7.4-cp310-cp310-macosx_12_0_arm64.whl`

**possible to open:**
`https://github.com/grekichi/p4a_wheel/raw/main/.github/ctc_segmentation-1.7.4-cp310-cp310-macosx_12_0_arm64.whl`

replace word 'blob' →  'raw'
I don't know why this is effective...

Anyway, wheel data build had been success.
Thanks for your cooperation.
